OutreachBot/src/main/java/frc/robot/subsystems/LEDSubsystem.java

```diff
@@ -0,0 +1,174 @@
+        m_userSelectedPatternChooser.addOption("gosLED", this::gosLED);
+        m_userSelectedPatternChooser.addOption("generalLEDTest", this::generalLEDTest);
+        m_userSelectedPatternChooser.addOption("gosColorTest", this::gosColorTest);
+        m_userSelectedPatternChooser.addOption("singlePixel", this::singlePixel);
+        m_userSelectedPatternChooser.addOption("fade", this::fade);
+
+        // TODO(PJ) Temporary debugging
+        m_userSelectedPatternChooser.addOption("limelightCheck(true)", () -> limelightCheck(true));
+        m_userSelectedPatternChooser.addOption("limelightCheck(false)", () -> limelightCheck(false));
+        m_userSelectedPatternChooser.addOption("shooterError(-20)", () -> shooterError(-20));
+        m_userSelectedPatternChooser.addOption("shooterError(-10)", () -> shooterError(-10));
+        m_userSelectedPatternChooser.addOption("shooterError(-5)", () -> shooterError(-5));
+        m_userSelectedPatternChooser.addOption("shooterError(1)", () -> shooterError(1));
+        m_userSelectedPatternChooser.addOption("shooterError(7)", () -> shooterError(7));
+        m_userSelectedPatternChooser.addOption("shooterError(11)", () -> shooterError(11));
+    }
+
+    public void rainbow() {
+        // For every pixel
+        for (var i = 0; i < m_ledBuffer.getLength(); i++) {
+            // Calculate the hue - hue is easier for rainbows because the color
+            // shape is a circle so only one value needs to precess
+            final var hue = (m_rainbowFirstPixelHue + (i * 180 / m_ledBuffer.getLength())) % 180;
+            // Set the value
+            m_ledBuffer.setHSV(i, hue, 255, 128);
+        }
+        // Increase by to make the rainbow "move"
+        m_rainbowFirstPixelHue += 3;
+        // Check bounds
+        m_rainbowFirstPixelHue %= 180;
+    }
+
+    public void gosLED() {
+        generalLED(0, NUMBER_LED / 2, 255, 0, 0);
+        generalLED(NUMBER_LED / 2, NUMBER_LED, 0, 0, 255);
+    }
+
+    //1st 4 blue, next 6 red, next 10 green, next 1 yellow, the rest purple
+    public void generalLEDTest() {
+        generalLED(0, 4, 0, 0, 255);
+        generalLED(4, 10, 255, 0, 0);
+        generalLED(10, 20, 0, 255, 0);
+        generalLED(20, 21, 255, 255, 0);
+        generalLED(21, NUMBER_LED, 153, 51, 255);
+    }
+
+    public void gosColorTest() {
+        for (var i = 0; i < NUMBER_LED; i += 2) {
+            m_ledBuffer.setRGB(i, 255, 0, 0);
+        }
+        for (var i = 1; i < NUMBER_LED; i += 2) {
+            m_ledBuffer.setRGB(i, 255, 255, 255);
+        }
+    }
+
+    public void singlePixel() {
+        generalLED(m_pixelCounter, m_pixelCounter + 1, 255, 0, 0);
+        if (m_loopCounter % 5 == 0) {
+            m_pixelCounter++;
+        }
+
+        if (m_pixelCounter >= NUMBER_LED) {
+            m_pixelCounter = 0;
+        }
+    }
+
+    public void clear() {
+        generalLED(0, NUMBER_LED, 0, 0, 0);
+    }
+
+    @SuppressWarnings("PMD")
+    public void fade() {
```

Comment:
I might call this `fadeGreen` or something. Lets a caller know it only works with one color in its current state